addons/web/static/src/views/fields/input_field_hook.js

```diff
@@ -145,7 +145,7 @@ export function useInputField(params) {
                 return;
             }
 
-            if (val !== component.props.value) {
+            if ((val || false) !== (component.props.value || false)) {
```

Comment:
In your commit message
> to prevent considering empty textarea string dirty.

I would add
> if the previous value was false (false and empty string should be considered as the same value)